### ISSUE TITLE
bl_pin ?

### ISSUE THREAD

**Issue by sjorge** [1coderookie/Klipper4KobraGoNeo#1]
I noticed both configs have bl_pin block, but it's not clear what this does?

```
[output_pin bl_pin]
pin: PC0
```

So, what does it do? Or is just just an artifact from copying the initial config from somewhere?

**Comment by 1coderookie** [comment]
Good question, you might wanna ask xpeng in the linked thread about that as his printer.cfg was the one we were using and adjusting to our needs.  
I *guess* that it's the ABL sensor tho, as "bl_pin" = bedlevel_pin would make sense in my opinion. 
However, it works fine for me and @cringegnere, but if you're going to ask xpeng, please share the answer here.

**Comment by sjorge** [comment]
Well it shows up as a toggle under misc in mainsail. But it does nothing and bed level seems to work regardless of the state.

Where can I reach out to xpeng ?

**Comment by 1coderookie** [comment]
You're right, never really thought about that one actually! :)

You find the link at my infopages (as mentioned in this repo as well) right in the first sentence of the "Installation" section of the "Klipper" chapter: https://1coderookie.github.io/KobraGoNeoInsights/firmware/fw_klipper/

**Comment by 1coderookie** [comment]
May I ask how you find this repo?

**Comment by sjorge** [comment]
I was looking for printer.cfg's to compare mine too, I  grabbed mine from reddit and already tuned it a bit more.
But couldn't figure out what the bl_pin was. (Or why my part fan isn't working, but they seem to be unrelated)

**Comment by 1coderookie** [comment]
Ah so you came here from the reddit post actually? I see.
If you wanna share your printer.cfg, please let me know, I'd be happy to add it as well! Guess we can all learn from other cfg-settings ;)
I might can tell you why your part cooling fan isn't working - most likely it's broken.. I already had two dying and other users reported this issue with the fan as well - and also with the stock firmware, so it doesn't seem to be related to Klipper actually.. :(

**Comment by sjorge** [comment]
> Ah so you came here from the reddit post actually? I see. If you wanna share your printer.cfg, please let me know, I'd be happy to add it as well! Guess we can all learn from other cfg-settings ;) I might can tell you why your part cooling fan isn't working - most likely it's broken.. I already had two dying and other users reported this issue with the fan as well - and also with the stock firmware, so it doesn't seem to be related to Klipper actually.. :(

I have about 200h of printing with Marlin with the fan, I have not had issues yet. But yes, I checked with marlin and it's broken. I replaced it, did a benchy on marlin, turned out 'fine' (as fine as I can get on marlin).

Went back to klipper, it won't work again, so I checked back on marlin and it was dead again. 

Maybe my fan config is wrong, but I did based it of xpeng's on reddit.

```
# Anycubic Kobra Neo

[include mainsail.cfg]
[include rpi.cfg]
[include bedmesh.cfg]
[include gcode.cfg]

[stepper_x]
step_pin: PA12
dir_pin: PA11
enable_pin: !PA15
microsteps: 16
rotation_distance: 40
position_endstop: -6
position_min: -6
position_max: 222
homing_speed: 60
endstop_pin: !PH2
homing_retract_dist: 0

[stepper_y]
step_pin: PA9
dir_pin: PA8
enable_pin: !PA15
microsteps: 16
rotation_distance: 40
position_endstop: -11
position_min: -11
position_max: 222
homing_speed: 60
endstop_pin: ^!PC13
homing_retract_dist: 0

[stepper_z]
step_pin: PC7
dir_pin: !PC6
enable_pin: !PA15
microsteps: 16
rotation_distance: 8
endstop_pin: ^PC14
position_min: -2
position_max: 250
position_endstop: 2
homing_speed: 5
second_homing_speed:1
homing_retract_dist: 2.3

[safe_z_home]
home_xy_position: 10, 10
speed: 65
z_hop: 5
z_hop_speed: 10

[extruder]
max_extrude_only_distance: 100.0
step_pin: PB15
dir_pin: !PB14
enable_pin: !PA15
microsteps: 16
rotation_distance: 7.60818
nozzle_diameter: 0.400
filament_diameter: 1.750
heater_pin: PB8
sensor_type: ATC Semitec 104GT-2
sensor_pin: PC3
min_temp: 5
min_extrude_temp: 5
max_temp: 270 #Stock is 250
control: pid
pid_Kp: 22.510
pid_Ki: 1.261
pid_Kd: 100.450

[firmware_retraction]
retract_length:1.3
retract_speed:35
unretract_extra_length:0
unretract_speed:30

[heater_bed]
heater_pin: PB9
sensor_type: EPCOS 100K B57560G104F
sensor_pin: PC1
min_temp: 0
max_temp: 130
control: pid
pid_Kp: 62.843
pid_Ki: 1.334
pid_Kd: 739.972

[bed_mesh]
speed: 120
horizontal_move_z: 5
mesh_min: 32, 32
mesh_max: 190, 190
algorithm: bicubic
probe_count: 8, 8

[probe]
pin: PA1
x_offset: 38.0
y_offset: 4.0
z_offset: 0.410
samples: 6
samples_result: average
samples_tolerance: 0.03
samples_tolerance_retries: 6
sample_retract_dist: 0.5
speed:1
lift_speed:4

[controller_fan controller_fan]
max_power: 0.75
pin: PB12

[heater_fan extruder_fan]
pin: PB13

# XXX: broken fan?
[fan]
pin: PB5

[output_pin enable_pin]
#This is the pin controls bed, hotend, extruder fan, part fan.
pin: PB6
static_value: 1

# XXX: what is this?
#[output_pin bl_pin]
#pin: PC0

#[output_pin tft_rst]
#pin: PB4

[idle_timeout]
timeout: 600

[mcu]
serial: /dev/serial/by-id/usb-1a86_USB_Serial-if00-port0
restart_method: command

[printer]
kinematics: cartesian
max_velocity: 500
max_accel: 2500
max_accel_to_decel: 2000
max_z_velocity: 20
max_z_accel: 20

[gcode_arcs]
```

**Comment by 1coderookie** [comment]
I also thought it might me the cfg, but referring to the marlin code the PB5 is correct. We also talked about this on reddit and xpeng(?) said it might be the PWM frequency, but he never had any problems and I set the power to 100% all the time so it might not be that as well. And as mentioned before, other users reported dead fans using stock-fw as well. So I asssume it's just bad quality actually, but maybe I'm wrong - if so, I hope someone will find the cause for this actually..

**Comment by sjorge** [comment]
I guess once the 3rd fan get here we'll know if it's something klipper related. I plan to do the following again:

- flash marlin
- connect display
- swap fan (but not installed)
- use display to test the fan (if OK continue)
- swap the actually fan out of the cable wrap (god this is such a pain)
- switch my my octoprint microSD in my pi
- print benchy (and watch fan closely)
- if success, switch back to klipper
- print bench (and watch fan closely)

If it died mid benchy on klipper like it did with the previous one, I'd say something would definitely be wrong

**Comment by 1coderookie** [comment]
Sounds like a plan, keep me updated plz as well as maybe also write here about it (here's the talk about the fan problem): https://www.reddit.com/r/anycubic/comments/10cwm16/comment/jair234/?utm_source=reddit&utm_medium=web2x&context=3
And i just saw your question about the pin there - so you're 1-w/o-0? Haha, nice to meet you here as well ;)

**Comment by sjorge** [comment]
Yes, will update here when I've done the above, and on reddit. And yes, that's me.

**Comment by sjorge** [comment]
In addition to some of the Anycubic replacement parts via 3djake I also just ordered a set of https://www.aliexpress.com/item/1005002065432445.html (24v ones) not sure which will be here faster. Not 100% sure they will fit, but if they do I might just switch to these as I can just run an extension cable to the hotend and have these swappable without undoing the cable harness.

And it gives me more fans in case I kill the anycubic replacement part again when trying klipper.

**Comment by 1coderookie** [comment]
Judging by the pictures of both they should fit imho, at least bot are 4510. Yeah, I need to place an order at 3DJ again as well to be on the safe side.. :( ;)

**Comment by sjorge** [comment]
The fan from 3dj arrive (so the exact same one), it spins on both klipper and marlin (With default `cycle_time). I'm gonna change the `cycle_time` to `0.130` and just wing it with klipper.

Given more users with a Neo showed up that had a sudden broken fan, I'm personally inclined to think it's a klipper thing that is not dialed in.

`cycle_time: 0.130` should roughtly close enough to 7.629hz which I believe from looking at the source from from Anycubic what the SOFT PWM is doing in Marlin.

Not sure if I'll have time to redo the  wire sleave as I just plugged it in and tested the fan with the display at 0, 25, 50, 100 and did the same thing via mainsail with the fan just on the shelve with the bottom of the printer open 😂 

Will post an update if I manage to complete a benchy

**Comment by 1coderookie** [comment]
Did you see my replies to yuo on reddit?
I'm using 10kHz since yesterday, before that I used 20Hz, at least until now (knock on wood) it's still working. I hope @jojos38 will come into the discussion as well as he has more knowledge about this whole fw stuff apparently than the two of us together lol (no offense tho! ;) )..

**Comment by 1coderookie** [comment]
Btw: I just noticed again the flickering underneath the printer of the green LED which I guess is caused by the fanspeed of the mainboard fan. Do you see that with the stock fw as well? If so, then it's imho definitely a sign that it's more likely spinning at the 7.7Hz than at a higher speed - I just wonder why it's visible then here with Klipper, if the default is 100Hz..?!

**Comment by sjorge** [comment]
I missed the post on reddit where you posted your new cycle_time. 

I think there was always a green flicker coming from underneath, but my memory is not the best.

**Comment by 1coderookie** [comment]
I actually meant what @jojos38 said about the frequency which I wrote underneath your latest post, assuming that it's about 31.4kHz.

**Comment by sjorge** [comment]
I can confirm that with `hardware_pwm: true` it doesn't turn on. `cycle_time: 0.032` seems to also work which would be `31.25hz`

**Comment by sjorge** [comment]
> > I can confirm that with `hardware_pwm: true` it doesn't turn on. `cycle_time: 0.032` seems to also work which would be `31.25hz`
> 
> I'm not a pro of Klipper but I think this should be the answer, I don't know how precise the cycle_time variable can be but the theoretical "Anycubic original value" would be 0.0318471

It seems to take 3 decimals so 0.032 was the closest I could get.

**Comment by 1coderookie** [comment]
Wait guys - 31.7**k**Hz isn't 31.7Hz - the one is 31700Hz, the other one 31.7Hz. So what is the actual value we were talking about @jojos38? You mentioned **k**Hz, not Hz. "But by using FAST_PWM_FAN and setting it to 31.4Khz you'll notice that the fan behaves exactly the same as the stock Anycubic firmware" - so 31.7**k**Hz would be more like 0.0000315s

**Comment by jojos38** [comment]
> Wait guys - 31.7**k**Hz isn't 31.7Hz - the one is 31700Hz, the other one 31.7Hz. So what is the actual value we were talking about @jojos38? You mentioned **k**Hz, not Hz. "But by using FAST_PWM_FAN and setting it to 31.4Khz you'll notice that the fan behaves exactly the same as the stock Anycubic firmware" - so 31.7**k**Hz would be more like 0.0000315s

Oh yes I'm sorry, the value should be 0.0000318471

On Marlin the stock behavior value is indeed 31400 Hz or 31.4 Khz

**Comment by 1coderookie** [comment]
Ok, thanks :) I'm in the middle of a print right now using 10kHz which produces a hight tone at 20%-60% it's printing with right now, but then I'll try to set it to the mentioned 31.4kHz. It also just leaves me kinda insecure and uncomfortable as it's mentioned in the Klipper docu that one should avoid using such high frequencies when using software PWM due to generating too much heat of the involved parts, but hardware PWM doesn't work somehow, so I guess I have to try it anyway..

**Comment by sjorge** [comment]
I don't think the resolution goes that high, but it's worth a shot.

Meanwhile, I took a picture of the board in the bad anycubic fan... sadly disassembly was more destructive than I had hoped but lots of glue.

![IMG_1684](https://user-images.githubusercontent.com/379665/225009479-1a642985-7bb9-4330-a3f5-08db77623d86.jpg)

**Comment by 1coderookie** [comment]
Thanks! So only that one IC.. I've read in an old thread of a german 3d printer forum that the problem with using (software) PWM with fans that aren't designed for (as they don't have the special 3rd/&4th wire) is, that due to the on/off/on/off the IC will most likely break. Some fans live longer if this occurs at a lower frequency, other fans live longer if it occurs at a higher frequency, which seems exactly to be the case here with our Neo.. Hopefully the manufacturer will reply, but tbh I don't think so.. :(

**Comment by 1coderookie** [comment]
Can you read whats printed on the IC? Like manufacturer, part number etc?

**Comment by 1coderookie** [comment]
Actually what I just spotted right now: see the "CC 4010" at the side? That's actually a fan model of that company! What's printed exactly there? Can't really see it..

**Comment by sjorge** [comment]
98% sure it says `CC-4010-J477`
2% sure it says `CC-4010-JA77`

**Comment by 1coderookie** [comment]
Ok, great, thanks, I'll look that up. 
Can you read the print on the iC itself as well?

**Comment by 1coderookie** [comment]
So I just set the 31.4**k**Hz = 0.000032 - definitely works, yound at various percentages is 'normal' like a fansound should be. When using the 0.130 for ~7.7Hz, it also works, but when reducing the speed with the slider to e.g. 50%, 60% and so on, I can hear more of a humming noise as well besides the fansound itself. Don't remember having heard that with the stock fw at any time actually..

**Comment by sjorge** [comment]
> Can you read the print on the iC itself as well?

Couldn't make out anything on the top, I tried to pull it off but I crushed it.

> I can hear more of a humming noise as well besides the fansound itself. Don't remember having heard that with the stock fw at any time actually..
The stock marlin to me at least sounded very clean, as in just fan noise.

**Comment by sjorge** [comment]
Something that popped into my head, does the MCU not run on a higher freq on klipper than on marlin ? I vaguely remember seeing something about that.

**Comment by 1coderookie** [comment]
Uhh, you broke it? Dammit :( Well, at least you didn't broke a working fan haha ;)
Yeah, same is what I remember..
So I just had another idea of an approach to maybe find the 'optimum' setting for the fan, despite the setting of the stock fw (as we saw broken fans there as well). 
So I looked up the specs for the 12V version of this model (http://www.coolcox.com/products/pdf/BF4510.pdf). Assuming that it's more or less the same rpm, it's 6000rpm then. If we now convert this to Hz, shouldn't we get the more or less ideal setting? Or am I completely wrong here?

**Comment by sjorge** [comment]
I'm not sure on the math to convert that, but according to coolcox's part number thing, the H is for the high speed version which all others do indeed seem to be around 6000rpm in that category.

**Comment by 1coderookie** [comment]
You mean the MCU itself? Like MHz wise of the 32bit processor? Dunno, never heard of that actually. But it's kinda weird anyway that the chip should be 200MHz and Mainsail shows 168MHz actually..

**Comment by sjorge** [comment]
> You mean the MCU itself? Like MHz wise of the 32bit processor? Dunno, never heard of that actually. But it's kinda weird anyway that the chip should be 200MHz and Mainsail shows 168MHz actually..

Yes, I remember something in the fork before it got merged about the chip normally running faster according to the specification but it was running at a slower speed.

**Comment by sjorge** [comment]
Used my break from work to start a calicube test print, with `cycle_time: 0.0000318471` it indeed sounds way better than with `0.130`.

Seems a bit louder than on marlin, but I did not reinstall the cover so that might be the reason.

**Comment by 1coderookie** [comment]
I was lazy and just took 0.000032 :)
Hopefully we're on the right way - the only things that still concerns me is the mentioned/supposed minimum value in the Klipper docs and that the fan also already died at other users Neo when using the stock fw.. But we'll see, at least you're on the safe side and have spare parts now, I still didn't order yet.. :(

**Comment by sjorge** [comment]
The spares are not the stock ones though, but hopefully they work too if this one dies (I hope not)

**Comment by 1coderookie** [comment]
Yeah, you mentioned that, but at least they fit as a replacement and you do have them already ;)

**Comment by sjorge** [comment]
> Yeah, you mentioned that, but at least they fit as a replacement and you do have them already ;)

And if this one dies quickly, at least here is a chance those will be less bad :D

**Comment by 1coderookie** [comment]
Never give up the hope - I like that ;)

**Comment by sjorge** [comment]
About 4h of printing with `cycle_time: 0.0000318471` so far no issues yet.

Well I made it through my cube, temp, retraction, retraction speed, fan speed, print speed towers and PA tuning pattern. Tomorrow a benchy with everything dialed in 🤞

**Comment by 1coderookie** [comment]
Sounds good! Also here that setting worked well so far, sounds kinda 'healthy' as well actually, so no 'weird' noises besides the actual fan sound. 
Pointing this out to @cringegnere as he also had a dead fan already (but when using stock fw iirc)..

**Comment by jojos38** [comment]
> Thanks! So only that one IC.. I've read in an old thread of a german 3d printer forum that the problem with using (software) PWM with fans that aren't designed for (as they don't have the special 3rd/&4th wire) is, that due to the on/off/on/off the IC will most likely break. Some fans live longer if this occurs at a lower frequency, other fans live longer if it occurs at a higher frequency, which seems exactly to be the case here with our Neo.. Hopefully the manufacturer will reply, but tbh I don't think so.. :(

I mean if the fan ever dies it's not extremely expensive to replace and you will probably get a better quality one on top of that



> About 4h of printing with `cycle_time: 0.0000318471` so far no issues yet.
> 
> Well I made it through my cube, temp, retraction, retraction speed, fan speed, print speed towers and PA tuning pattern. Tomorrow a benchy with everything dialed in 🤞

Awesome then!

**Comment by jojos38** [comment]
> Ok, thanks :) I'm in the middle of a print right now using 10kHz which produces a hight tone at 20%-60% it's printing with right now, but then I'll try to set it to the mentioned 31.4kHz. It also just leaves me kinda insecure and uncomfortable as it's mentioned in the Klipper docu that one should avoid using such high frequencies when using software PWM due to generating too much heat of the involved parts, but hardware PWM doesn't work somehow, so I guess I have to try it anyway..

Also yes this is true, higher hardware PWM increases heat and can even burn the motherboard but I have no idea at what point it becomes dangerous

**Comment by 1coderookie** [comment]
> I mean if the fan ever dies it's not extremely expensive to replace and you will probably get a better quality one on top of that
> 

True. It's just that this 4510 24V type isn't *that* common and easy to get (at least here in Germany) and besides that it's just kinda 'meh' to keep a handful of part cooling fans as spare parts in the drawer as well. So if we manage to find a solution here it would be the best of course.  
However, I'm happy that we're trying to solve this issue together instead of just one person on his own - highly appreciate it! :)

Besides that, I still keep looking for fanducts which fit a 5015 radial blower and which may fit our Neo, ideally w/o the need of a whole new construction (as I'm totally unexperienced with creating models). I *think* the extruder system of the Neo is actually some kind of clone of the Titan Aero, so I found this https://www.thingiverse.com/thing:2876716 and maybe will give it a try and tesprint it, just to see if it may be a good base model to adapt to the Neo.

> Also yes, higher software PWM increases heat and can even damage the motherboard but I have no idea at what point it becomes dangerous

Yeah, that's my fear actually, as I already replaced the mainboard one time and really don't wanna spend money on that again (right now). But *if* I'd had to do that, I probably would go for an SKR/BTT with independent dual z-steppers anyway - just wasn't into this whole topic right at the beginning when I swapped out the broken mainboard.. :( ;)

**Comment by 1coderookie** [comment]
Actually I'm thinking about if it would be possible to use e.g. a Pico as a secondary MCU for connecting and controlling the part cooling fan.. Guess I'll do some research on that as well, as it may be a cheap 'protection' of the mainboard if something would blow up due to the 31.4kHz..

**Comment by sjorge** [comment]
> Actually I'm thinking about if it would be possible to use e.g. a Pico as a secondary MCU for connecting and controlling the part cooling fan.. Guess I'll do some research on that as well, as it may be a cheap 'protection' of the mainboard if something would blow up due to the 31.4kHz..

Not everyone that wants to run klipper might be willing or tech savy enough to do that though. 

Are we not doing the same thing anycubic marlin fw is doing? So the later would cause the same wear on the board right? Unless that is indeed somehow using hardware pwm and thats not working on klipper yet?

**Comment by sjorge** [comment]
I wasted a 1 whole spoon on marlin until were I was sort of OK with the result but it was still not how I wanted it, it was better than what I'm getting atm though -_-

**Comment by 1coderookie** [comment]
Oh my.. Better than at the moment? Whaaat?!?

**Comment by sjorge** [comment]
> Oh my.. Better than at the moment? Whaaat?!?

Yeah, I got it pretty OK aside from small blobbing near the z-seam, so I started messing with the marlin source but never got linear advance to work (it would stall the extruder at around Z layer 5-10).

At least pressure advance is working on klipper.

**Comment by 1coderookie** [comment]
Yeah, PA doesn't really seem to work with the 2208, I guess that's actually one of the mainreasons ppl switch to Klipper.
I'm actually struggling with stringing again after I raised the temp up to 220° as it seems to give better results stability wise, and no matter what I set, it just doesn't get better/away. Plus annoying holes between lines at the top layers as well, even tho I also use Arachne and tuned the settings for top layer, flow etc. as well.. 
So annoying..

**Comment by sjorge** [comment]
For PLA (Anycubic and eSUN) I used:
- retraction speed: 40
- retraction distance: 0.88
- print temp: 200-205

Stringing has been decent with those, I have more with a retraction distance of 1 than 0.8.  This is with a 0.4mm nozzle though (A Bontech CHT), I think you're using 0.6mm so might not be comparable at all.

A drybox does help too btw, but i only use that for PETG

**Comment by 1coderookie** [comment]
Thanks! Pretty similar to my settings when I printed with 200°, I'm using "AzureFilm Strongman PLA" with a 0.6mm, right. Right now with 220° I'm at 0.8mm r.distance with 50mm r.speed, so I'll try to go up to 0.9-1.0mm I guess. Retraction tower doesn't really show a difference, which is kinda weird, so I probably overlook something here as well, dunno.. 
What did you set as de-retraction speed? I'm at 35-40mm/s right now, but not even sure if that would make any difference actually haha

Drybox: I built a ball bearing spool holder from a cakebox (mounted to a shelf above the printer) with a big bag of silica inside to keep it as dry as possible, only made a tiny hole in it of course for guiding the filament thru. Filament comes out the nozzle homogen and nice actually, so I really think the 220° are more the problem here.. But for longterm I should get myself one of those fruit driers probably..

**Comment by 1coderookie** [comment]
Ha, wait - that may be it! So if you're using these settings with a 0.4mm nozzle and I'm using 0.6mm, then it's maybe already the cause, cuz the diameter is 'too' big! THANKS, never brought my nozzle diameter into account when comparing to other settings in this case actually!

**Comment by sjorge** [comment]
I have my detraction speed the same as my retraction speed.

I got one of the eSUN ones from amazon when they were on sale over christmas, was like 35 EUR or something like that.

**Comment by sjorge** [comment]
> Sorry if I'm interrupting this thread at the wrong time, but which exact fan are you guys buying to replace the part cooling fan? It seems mine has died as well. And what does the cycle_time have to be set at?

I got the exact same one that came with the printer on 3djake.be.

https://www.3djake.be/nl-BE/anycubic-3d-printers-accessoires/material-fan-2

Make sure to pick the correct one, as the cable needs to be long.

**Comment by LoiLock** [comment]
And just to be 100% sure, `cycle_time: 0.0000318471` is the one that's been working for people? If this is the case, it should probably be reflected in the config files in the repo.

**Comment by sjorge** [comment]
> And just to be 100% sure, `cycle_time: 0.0000318471` is the one that's been working for people? If this is the case, it should probably be reflected in the config files in the repo.

Been working for me, about 12h of prints on it now (mostly towers for calibration).

Here is the printer.cfg I was using last night
```ini
##
## Anycubic Kobra Neo
##
[stepper_x]
step_pin: PA12
dir_pin: PA11
enable_pin: !PA15
microsteps: 16
rotation_distance: 40
position_endstop: -6
position_min: -6
position_max: 222
homing_speed: 60
endstop_pin: !PH2
homing_retract_dist: 0

[stepper_y]
step_pin: PA9
dir_pin: PA8
enable_pin: !PA15
microsteps: 16
rotation_distance: 40
position_endstop: -11
position_min: -11
position_max: 222
homing_speed: 60
endstop_pin: ^!PC13
homing_retract_dist: 0

[stepper_z]
step_pin: PC7
dir_pin: !PC6
enable_pin: !PA15
microsteps: 16
rotation_distance: 8
endstop_pin: ^PC14
position_min: -2
position_max: 250
position_endstop: 6.5
homing_speed: 5
second_homing_speed:1
homing_retract_dist: 2.3

[safe_z_home]
home_xy_position: 110, 110
speed: 65
z_hop: 5  
z_hop_speed: 10

[extruder]
step_pin: PB15
dir_pin: !PB14
enable_pin: !PA15
microsteps: 16
full_steps_per_rotation: 200
rotation_distance: 7.60818
nozzle_diameter: 0.400
filament_diameter: 1.750
max_extrude_only_distance: 100
max_extrude_only_accel: 100
heater_pin: PB8
sensor_type: ATC Semitec 104GT-2
sensor_pin: PC3
min_temp: 5
min_extrude_temp: 5
max_temp: 270 # Stock is 250
control: pid
pid_Kp: 21.895
pid_Ki: 1.315
pid_Kd: 91.136

[firmware_retraction]
retract_length:1.3
retract_speed:35
unretract_extra_length:0
unretract_speed:30

[heater_bed]
heater_pin: PB9
sensor_type: EPCOS 100K B57560G104F
sensor_pin: PC1
min_temp: 0
max_temp: 130
control: pid
pid_Kp: 59.972
pid_Ki: 0.917
pid_Kd: 980.543

[bed_mesh]
speed: 120
horizontal_move_z: 5
mesh_min: 32, 32
mesh_max: 190, 190
algorithm: bicubic
probe_count: 8, 8

[probe]
pin: PA1
x_offset: 38.0
y_offset: 4.0
z_offset: 0.156
samples: 6
samples_result: average
samples_tolerance: 0.03
samples_tolerance_retries: 6
sample_retract_dist: 0.5
speed: 1
lift_speed: 4
activate_gcode: PROBE_RESET

[screws_tilt_adjust]
screw1: -6, 28
screw1_name: front left screw
screw2: 158, 28
screw2_name: front right screw
screw3: 158, 193
screw3_name: rear right screw
screw4: -6, 193
screw4_name: rear left screw
horizontal_move_z: 10
speed: 50
screw_thread: CCW-M4

[controller_fan controller_fan]
max_power: 0.75
pin: PB12
stepper: stepper_x,stepper_y,stepper_z
heater: heater_bed

[heater_fan extruder_fan]
pin: PB13

[fan]
pin: PB5
#cycle_time: 0.050 # slipeete 20hz
cycle_time: 0.0000318471 # jojos38 31.4kHz ???

[output_pin enable_pin]
#This is the pin controls bed, hotend, extruder fan, part fan.
pin: PB6
static_value: 1

[output_pin probe_reset_pin] 
pin: PC0
value: 1

[gcode_macro PROBE_RESET]
gcode:
  SET_PIN PIN=probe_reset_pin VALUE=0
  G4 P300
  SET_PIN PIN=probe_reset_pin VALUE=1
  G4 P100

#[output_pin tft_rst]
#pin: PB4

[idle_timeout]
timeout: 600

[mcu]
serial: /dev/serial/by-id/usb-1a86_USB_Serial-if00-port0
restart_method: command

[printer]
kinematics: cartesian
max_velocity: 500
max_accel: 500
max_z_velocity: 25
max_z_accel: 100
max_z_accel: 20
```

I'm experimenting with a few things, so you might just want to grab the `[fan]` and `[controller_fan controller_fan]` sections.

Additional changes:
- acceleration, I copied the settings from KorbaNeo firmware drop from Anycubic, these are much more conservative
- bl_pin -> renamed probe_reset_pin as it seems other Anycubic example klipper configs all have one of those.
- screws_tilt_adjust (if you try this, adjust the screw location offsets as they are xy endstop dependant)

**Comment by 1coderookie** [comment]
> Edit: @1coderookie perhaps a link on the site should be added to the exact replacement fan? Seems 3djake ships almost everywhere.

Yeah, I had 3DJ mentioned as a good supplier in the Spare Parts chapter "Where To Get?", but I deleted it again actually as I might get in trouble adding links which could be seen as affiliate links by certain laws here, cuz I don't have (and don't want to add ;) ) an imprint (which I don't need if it's a pure personal and non-commercial site). So I'm actually torn between mentioning it or just leave it to the people to search for the mentioned part. There's a picture of the fan anyway, so ppl can see the exact name of the part..

**Comment by 1coderookie** [comment]
> And just to be 100% sure, `cycle_time: 0.0000318471` is the one that's been working for people? If this is the case, it should probably be reflected in the config files in the repo.

We're still investigating and testing that, so there's no *guarantee* that it's correct. 
So for my own config in this repo: yes, I'll upload an updated version soon. 
For @cringegnere 's config: I don't wanna mess with that as it's not mine. 
For the other guys configs which I linked to: there's nothing I can do about it actually. I/we also talked about this in the initial reddit post, but somehow nobody really responds there, so we're actually talking about it here.. 
I personally put in 0.000032, which is 31.25kHz, as I was lazy and didn't wanna stress Klipper with so many numbers.. ;)

**Comment by 1coderookie** [comment]
CCW: Well ok, gotta try that again then next time - as I said, somehow neither one gave me the actually correct instruction for all three screws, so that was kinda weird.. ;)

**Comment by 1coderookie** [comment]
> > Did you already do the Input Shaper config?
> 
> Not yet, some day.

Then do yourself a favour and do by it using an ADXL345 - it's just sooo comfortable ;)
